### PR TITLE
Backup: add S3 metrics && add s3_multi_part_size config 

### PR DESCRIPTION
**Pull request by 3pointer**
### What is changed and how it works?

Part solve https://github.com/pingcap/tidb/issues/30087

What's Changed:
1. Add metrics for backup s3 storage. to record the qps that BR send to S3. 
2. BR use multi_upload when put a big object to S3. and the split size is 5MB. which may generate too many requests. This PR tried to make the multi_part_size configurable.

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test
- Manual test (add detailed scripts or steps below)

### Release note 

```release-note
Add a new hidden config `s3_multi_part_size` to make backup can control the part size when upload big sst file to s3.
```

### CODE REVIEW

**Review by YuJuncen — APPROVED** [review]
lgtm

File: components/cloud/aws/src/metrics.rs [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+use lazy_static::*;
+use prometheus::*;
+
+lazy_static! {
+    pub static ref S3_REQUEST_HISTOGRAM_VEC: HistogramVec = register_histogram_vec!(
+        "s3_request_duration_seconds",
+        "Bucketed histogram of s3 requests duration",
+        &["type"]
+    )
```
Comment (kennytm) [inline components/cloud/aws/src/metrics.rs:11]: 1. looks strange that the metrics is s3-only. there could be a new label (schema/provider/whatever) with value "s3".
2. the metrics name should be namespaced like `tikv_cloud_…`

**Review by kennytm — COMMENTED** [review]
rest lgtm

File: components/cloud/aws/src/metrics.rs (hunk 1 above)
Comment (3pointer) [reply components/cloud/aws/src/metrics.rs]: Done, PTAL again~ @kennytm

**Review by kennytm — APPROVED** [review]

**Review by Little-Wallace — APPROVED** [review]
LGTM

**Comment by kennytm** [comment]
@3pointer need to fix that `[do-not-merge/invalid-commit-message]` error first.

**Comment by kennytm** [comment]
ping @3pointer

**Pull request merged**